bindings/python/src/sha1_hash.cpp

```diff
@@ -33,7 +33,7 @@ void bind_sha1_hash()
         .def(init<std::string>())
         .def("clear", &sha1_hash::clear)
         .def("is_all_zeros", &sha1_hash::is_all_zeros)
-        .def("to_string", &sha1_hash::to_string)
+        .def("to_string", sha1_hash_bytes)
```

Comment:
This would break any current callers, which is probably fine since it rarely worked. But why not just delete it instead?

Reply:
I think leaving it in preserves fidelity with the C++ interface. since there isn't really any python specific API documentation, staying similar to C++ is helpful.

both removing and changing the return type would break current callers.

Reply:
I see.

If I said I had some thoughts on "documenting" the python API in the future, would it sway you? :) This is a minor point and I'm not willing to die on this hill, but I'm compelled to _try_, since the upside is avoiding extraneous APIs.

Check this out:

https://github.com/AllSeeingEyeTolledEweSew/tvaf/blob/master/typeshed/libtorrent.pyi

Modern python (3.6+) supports type annotations and type checking. For extensions like libtorrent, you can write "type stub" files which just have type annotations. They're header files for python.

I wrote a local `libtorrent.pyi` (against RC_1_2) to help with type-checking my own code. My end goal is to have it incorporated into libtorrent.

But in using type-checked python, I have found it extremely useful to treat type stubs _as documentation_. The type stubs of the standard libraries are an excellent quick reference, and half the time I just refer to those instead of referring to the official documentation pages.

I don't expect libtorrent's python bindings will ever get official documentation, but if they included official type stubs I would strongly argue we don't need strong fidelity with the C++ interface.

Reply:
> but if they included official type stubs I would strongly argue we don't need strong fidelity with the C++ interface.

sounds reasonable